src/main/java/com/mineboundteam/minebound/entity/MyrialSwordEntity.java

```diff
@@ -0,0 +1,93 @@
+package com.mineboundteam.minebound.entity;
+
+import com.mineboundteam.minebound.entity.registry.EntityRegistry;
+import com.mineboundteam.minebound.item.registry.ItemRegistry;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.projectile.ProjectileUtil;
+import net.minecraft.world.entity.projectile.ThrowableItemProjectile;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.EntityHitResult;
+import net.minecraft.world.phys.HitResult;
+import net.minecraft.world.phys.Vec3;
+import org.jetbrains.annotations.NotNull;
+
+public class MyrialSwordEntity extends ThrowableItemProjectile {
+    int maxEntitiesHit = 16;
+    IntOpenHashSet entitiesHit = new IntOpenHashSet(maxEntitiesHit);
+    ItemStack myrialSwordVacuum;
+
+    public MyrialSwordEntity(EntityType<MyrialSwordEntity> entityType, Level level) {
+        super(entityType, level);
+    }
+
+    public MyrialSwordEntity(Player player, Level level, ItemStack myrialSwordVacuum) {
+        super(EntityRegistry.MYRIAL_SWORD_ENTITY.get(), player, level);
+        this.myrialSwordVacuum = myrialSwordVacuum;
+    }
+
+    @Override
+    public boolean canHitEntity(@NotNull Entity entity) {
+        return !entitiesHit.contains(entity.getId());
+    }
+
+    @Override
+    protected @NotNull Item getDefaultItem() {
+        return ItemRegistry.MYRIAL_SWORD.get();
+    }
+
+    @Override
+    public boolean isInWater() {
+        return false;
+    }
+
+    @Override
+    public boolean isNoGravity() {
+        return true;
+    }
+
+    @Override
+    public void onHitEntity(@NotNull EntityHitResult entityHitResult) {
+        if (getOwner() != null && entityHitResult.getEntity().is(getOwner())) {
+            discard();
+            switchToMyrialSword((Player) getOwner());
+        } else {
+            entityHitResult.getEntity().hurt(DamageSource.MAGIC, 7);
+            entitiesHit.add(entityHitResult.getEntity().getId());
+        }
+    }
+
+    @Override
+    public void tick() {
```

Comment:
Also there should probably be a failsafe for when returning in case it can't get to the player after a few seconds